### PR TITLE
Remove unused roles

### PR DESCRIPTION
**Pull request by mpolden**
@jonmv

FYI @tokle @smorgrav

### CODE REVIEW

File: controller-api/src/main/java/com/yahoo/vespa/hosted/controller/api/integration/user/Roles.java [hunk 1]
```diff
@@ -57,10 +54,6 @@ public static Role toRole(TenantName tenant, String roleName) {
     /** Returns the {@link Role} the given tenant and role names correspond to. */
     public static Role toRole(TenantName tenant, ApplicationName application, String roleName) {
         switch (roleName) {
-            case "applicationAdmin": return Role.applicationAdmin(tenant, application);
-            case "applicationOperator": return Role.applicationOperator(tenant, application);
-            case "applicationDeveloper": return Role.applicationDeveloper(tenant, application);
-            case "applicationReader": return Role.applicationReader(tenant, application);
```
Comment (jonmv) [inline controller-api/src/main/java/com/yahoo/vespa/hosted/controller/api/integration/user/Roles.java]: Will the potential exception be caught by the caller?

Comment (mpolden) [reply controller-api/src/main/java/com/yahoo/vespa/hosted/controller/api/integration/user/Roles.java]: Yes, see internal PR.

File: controller-api/src/main/java/com/yahoo/vespa/hosted/controller/api/role/Role.java [hunk 2]
```diff
@@ -43,31 +43,16 @@ public static TenantRole athenzTenantAdmin(TenantName tenant) {
         return new TenantRole(RoleDefinition.athenzTenantAdmin, tenant);
     }
 
-    /** Returns a {@link RoleDefinition#tenantPipeline} for the current system and given tenant and application. */
-    public static ApplicationRole tenantPipeline(TenantName tenant, ApplicationName application) {
-        return new ApplicationRole(RoleDefinition.tenantPipeline, tenant, application);
+    /** Returns a {@link RoleDefinition#integrationTest} for the current system and given tenant and application. */
+    public static ApplicationRole integrationTest(TenantName tenant, ApplicationName application) {
+        return new ApplicationRole(RoleDefinition.integrationTest, tenant, application);
```
Comment (jonmv) [inline controller-api/src/main/java/com/yahoo/vespa/hosted/controller/api/role/Role.java]: On second thought, the integration test is a `hostedOperator`, so this isn't really needed.

**Review by jonmv — APPROVED** [review]
Great!

**Comment by jonmv** [comment]
The whole `InstanceRole` is also unused, and can be deleted.

**Comment by mpolden** [comment]
PTAL

**Review by jonmv — APPROVED** [review]

**Pull request merged**